File: tests/XMP/global-view/util/F/sub/sub_mpi.f90
subroutine sub_mpi(comm)
  include 'mpif.h'
  integer comm, rank, irank

  call MPI_Comm_rank(MPI_COMM_WORLD, rank, ierr)
  call MPI_Comm_rank(comm, irank, ierr)

  if(rank == 2) then
    if(irank == 1) then
      print *,"PASS"
    else
      print *, "ERROR rank=",irank
      call exit(1)
    end if
  end if

end subroutine

File: tests/XMP/global-view/util/C-Square/sub/sub_mpi.c
#include <stdio.h>
#include <stdlib.h>
#include "mpi.h"

int sub_mpi(MPI_Comm *comm) {

  int rank, irank;

  MPI_Comm_rank(MPI_COMM_WORLD, &rank);

  if (rank == 2){
    MPI_Comm_rank(*comm, &irank);
    if(irank == 1){
      printf("PASS\n");
    }
    else{
      fprintf(stderr, "ERROR rank=%d\n",irank);
      exit(1);
    }
  }

  return 0;
}
